saleor/product/utils/costs.py

```diff
@@ -58,5 +58,9 @@ def get_margin_for_variant(variant):
         return None
     base_price = variant.base_price
     margin = base_price - variant.cost_price
-    percent = round((margin / base_price) * 100, 0)
+    try:
```

Comment:
Hey @MahmoudRizk, thanks for opening the PR! I think it would make the most sense if we returned `None` in that case, similarly as we do if `cost_price` is also undefined. So the only change I would make is this:

```
base_price = variant.base_price
if not base_price:
    return None
```